anvil/src/config.rs

```diff
@@ -323,19 +323,19 @@ impl NodeConfig {
 
     /// Sets the chain ID
     #[must_use]
-    pub fn with_chain_id<U: Into<u64>>(mut self, chain_id: U) -> Self {
-        self.set_chain_id(chain_id.into());
+    pub fn with_chain_id<U: Into<u64>>(mut self, chain_id: Option<U>) -> Self {
+        self.set_chain_id(chain_id);
         self
     }
 
     /// Sets the chain id and updates all wallets
-    pub fn set_chain_id(&mut self, chain_id: impl Into<u64>) {
-        self.chain_id = chain_id.into();
+    pub fn set_chain_id(&mut self, chain_id: Option<impl Into<u64>>) {
+        self.chain_id = chain_id.map(Into::into);
         self.genesis_accounts.iter_mut().for_each(|wallet| {
-            *wallet = wallet.clone().with_chain_id(self.chain_id);
+            *wallet = wallet.clone().with_chain_id(self.chain_id.unwrap_or(CHAIN_ID));
```

Comment:
instead of using `self.chain_id.unwrap_or(CHAIN_ID)` everywhere, let's add another getter, see `get_gas_price` for example

Reply:
Ok will add another getter